addons/io_scene_gltf2/blender/exp/gltf2_blender_gather_animation_sampler_keyframes.py

```diff
@@ -325,6 +325,12 @@ def gather_keyframes(blender_object_if_armature: typing.Optional[bpy.types.Objec
         else: # bone is not animated (no fcurve)
             # Not keeping if not changing property
             return None if std < 0.0001 else keyframes
+    else:
+        # For objects, if all values are the same, we keep only first and last
+        # TODO We need to differenciate morph anim / TRS anim
+        std = np.ptp(np.ptp([[k.value[i] for i in range(len(keyframes[0].value))] for k in keyframes], axis=0))
+        return [keyframes[0], keyframes[-1]] if std < 0.0001 and len(keyframes) >= 2 else keyframes
```

Comment:
```py
ptps = np.ptp([[k.value[i] for i in range(len(keyframes[0].value))] for k in keyframes], axis=0)
```

This is an array of the ptp for each component. It's zero where one component is constant over time.

```py
np.ptp(ptps) < 0.0001
```

This part is wrong. It checks if the ptp of the ptps is zero, ie. all the ptps are the same. It should be checking if all the ptps are (approximately) zero. For example

```py
np.ptp(np.ptp( [[0,4], [3, 7]], axis=0))
```

is zero but the curves aren't constant.

Reply:
Seems #1433 had the same problem. Why don't you pull this "is constant" check out into a function?

Reply:
Seems #1433 had the same problem. Why don't you pull this "is constant" check out into a function?

Reply:
Thanks for your feedback. 
I got the same conclusion during my tests.